test_util/cluster_api.py

```diff
@@ -241,280 +241,43 @@ def __init__(self, dcos_uri, masters, public_masters, slaves, public_slaves,
-                       else Endpoint(t['host'], 0, t['ipAddresses'][0]['ipAddress'])
-                       for t in data['app']['tasks']]
-                logging.info('Application deployed, running on {}'.format(res))
-                return res
-            elif (not check_tasks_running):
-                logging.info('Waiting for application to be deployed: '
-                             'Not all instances are running: {}'.format(repr(data)))
-                return None
-            elif (not check_tasks_healthy):
-                logging.info('Waiting for application to be deployed: '
-                             'Not all instances are healthy: {}'.format(repr(data)))
-                return None
-            else:
-                logging.info('Waiting for application to be deployed: {}'.format(repr(data)))
-                return None
-
-        try:
-            return _pool_for_marathon_app(app_definition['id'])
-        except retrying.RetryError:
-            raise Exception("Application deployment failed - operation was not "
-                            "completed in {} seconds.".format(timeout))
-
-    def destroy_marathon_app(self, app_name, timeout=120):
-        """Remove a marathon app
-
-        Abort the test if the removal was unsuccesful.
-
-        Args:
-            app_name: name of the applicatoin to remove
-            timeout: seconds to wait for destruction before failing test
-        """
-        @retrying.retry(wait_fixed=1000, stop_max_delay=timeout * 1000,
-                        retry_on_result=lambda ret: not ret,
-                        retry_on_exception=lambda x: False)
-        def _destroy_complete(deployment_id):
-            r = self.get('/marathon/v2/deployments', headers=self._marathon_req_headers())
-            assert r.ok
-
-            for deployment in r.json():
-                if deployment_id == deployment.get('id'):
-                    logging.info('Waiting for application to be destroyed')
-                    return False
-            logging.info('Application destroyed')
-            return True
-
-        r = self.delete('/marathon/v2/apps' + app_name, headers=self._marathon_req_headers())
-        assert r.ok
-
-        try:
-            _destroy_complete(r.json()['deploymentId'])
-        except retrying.RetryError:
-            raise Exception("Application destroy failed - operation was not "
-                            "completed in {} seconds.".format(timeout))
-
-    @contextmanager
-    def marathon_deploy_and_cleanup(self, app_definition, timeout=120, check_health=True, ignore_failed_tasks=False):
-        yield self.deploy_marathon_app(
-            app_definition, timeout, check_health, ignore_failed_tasks)
-        self.destroy_marathon_app(app_definition['id'], timeout)
+            raise Exception('Node {} not found in DC/OS cluster'.format(node))
+
+    def get_client(self, path, default_headers=None):
+        if default_headers is None:
```

Comment:
looking at this again, ApiClient already does the None -> dict() logic, so doing nothing here and just letting the None fall through into ApiClient's __init__ is plenty